ampd/Dockerfile

```diff
@@ -1,10 +1,32 @@
-FROM rust:bookworm as build
-WORKDIR /app
+FROM rust:1.74-bookworm as builder
 RUN apt-get update && apt-get install -y clang protobuf-compiler
-COPY . .
+
+WORKDIR /ampd
+COPY ./Cargo.toml ./Cargo.toml
+COPY ./ampd/Cargo.toml ./ampd/Cargo.toml
+COPY ./ampd/Cargo.lock ./ampd/Cargo.lock
+COPY ./packages ./packages
+COPY ./contracts ./contracts
+COPY ./integration-tests ./integration-tests
+COPY ./.cargo ./.cargo
+
+# build dependencies separately
+RUN mkdir ampd/src && echo 'fn main() {}' > ampd/src/main.rs
 RUN cargo build --release
 
-FROM debian:bookworm-slim AS final
-COPY --from=build /app/target/release/ampd /usr/local/bin/ampd
-WORKDIR /usr/local/bin
-ENTRYPOINT ["ampd"]
+COPY ./ampd/proto ./ampd/proto
+COPY ./ampd/src ./ampd/src
+COPY ./ampd/build.rs ./ampd/build.rs
+
+RUN rustup component add rustfmt
```

Comment:
why is the formatter needed for release ?

Reply:
It doesn't, removed.  This was copied over from tnfnd dockerfile.